File: typescript/src/nuc/BLOCK.ts
import { $ } from "../graph";
import Instruction from "../Instruction";
import NODE from "./NODE";
import { v4 as uuid } from "uuid";

class BLOCK extends NODE {
  public statements: any[];
  public skip: boolean;

  constructor(key) {
    super(key);
    this.statements = [];
    this.skip = false;
  }

  run(scope: any): any {
    return this.statements.map(
      (statement, index) =>
        new Instruction(scope, statement, null, null, null, null)
    );
  }
}

export default BLOCK;

File: cpp/src/nuc/BLOCK.cpp
#include "nucleoid/nuc/BLOCK.hpp"

namespace nucleoid {

BLOCK::BLOCK(const std::string& key) : NODE(key) { type = "BLOCK"; }

void BLOCK::before(Scope* /*scope*/) {}
NODE* BLOCK::run(Scope* /*scope*/) { return nullptr; }
void BLOCK::graph(Scope* /*scope*/) {}
void BLOCK::after(Scope* /*scope*/) {}

void BLOCK::stage(const Instruction& instr) {
    staged_.push_back(instr);
}

} // namespace nucleoid
